### DISCUSSION TITLE
issue with rotating and dragging of rectangle with an angle

### DISCUSSION THREAD

**Discussion by anujs099** [geoman-io/leaflet-geoman#1510] [Questions & Help]
Hey guys, need help .

i am using react-leaflet to draw rectangle and updating its bounds every time i edit , drag or rotate using geo man tool. how can i make sure that the rectangle maintains that angle when i am performing any of three operations ? currently when i edit or drag after rotating the rectangle it goes back to its original angle.

@Falke-Design

**Comment by Falke-Design** [comment]
For me it sounds like you don't update the coordinates after dragging / editing and then because of reacts state management, it updates it back to before the edit.

You can try to store the latlngs with each `pm:change` event.
